app/services/status_update_service.rb

```diff
@@ -32,7 +32,7 @@ def call
 
     if delivery_attempt.permanent_failure? && subscriber
       UnsubscribeService.subscriber!(subscriber, :non_existant_email)
-    elsif delivery_attempt.temporary_failure?
+    elsif delivery_attempt.temporary_failure? && email.pending?
```

Comment:
Why do we need this extra check here?

Reply:
Because they may have exhausted retries and email could be failed.

Reply:
Oh yes sorry I was getting confused and thinking we were dealing with permanent failures when doing the retries exhausted state.